radicle-cli/src/commands/push.rs

```diff
@@ -0,0 +1,134 @@
+use std::ffi::OsString;
+use std::path::Path;
+
+use radicle::git;
+
+use crate::terminal as term;
+use crate::terminal::args::{Args, Error, Help};
+
+use anyhow::anyhow;
+
+pub const HELP: Help = Help {
+    name: "push",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad push [--all] [--[no-]sync] [<option>...]
+
+    By default, only the current branch is synced.
+
+Options
+
+    --all               Push all branches (default: false)
+    --sync              Sync after pushing to the "rad" remote (default: false)
+    --no-sync           Do not sync after pushing to the "rad" remote
```

Comment:
Why have both `sync` and `no-sync` flags? :thinking: Seems weird to have and feels like it should have precedence rules if it they both stay, i.e. which one trumps the other?

Reply:
Good question - I usually have a `--no-*` version for many boolean flags as I think it's useful as self-documentation when you see the command written and to not force users to remember what the default is.

As far as I can tell this is pretty standard, eg. see this screenshot from the ripgrep manual:
<img width="662" alt="image" src="https://user-images.githubusercontent.com/40774/201991457-20f28095-dc31-41eb-962f-917f3708c190.png">

Reply:
Ya, so maybe we should follow suit and make `--no-sync` override any use of `--sync`.

Reply:
https://github.com/radicle-dev/heartwood/issues/106